QueryGenie/CoreData/CoreDataQueryable.swift

```diff
@@ -8,9 +8,7 @@ import Foundation
 import CoreData
 
 /// A queryable entity that executes an `NSFetchRequest`.
-public protocol CoreDataQueryable: GenericQueryable, Enumerable {
-    
-    associatedtype Element: NSFetchRequestResult
+public protocol CoreDataQueryable: Queryable, Enumerable where Element: NSFetchRequestResult {
```

Comment:
Since we are getting rid of `GenericQueryable` this actually needs a new `associatedType` for the `NSFetchRequestResult`. The `Element` we are querying may not be the same type as the result.